### ISSUE TITLE
Error occured when setting the environment

### ISSUE THREAD

**Issue by TTTanger** [xdit-project/xDiT#388]
I am deploying the project on my **Linux** server, **Ubuntu 24.04.1 LTS**.
**Python version: 3.12.3**
Firstly, I used **"python3 -m venv .venv"**, **"source .venv/bin/activate"** to set the right working env, and I followed the _Quickstart_ to build the dependencies, but it occured:
```
Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      Traceback (most recent call last):
        File "/dk1/xDiT/.venv/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/dk1/xDiT/.venv/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/dk1/xDiT/.venv/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-upe2dpqk/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 334, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-upe2dpqk/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 304, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-upe2dpqk/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 522, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-upe2dpqk/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 320, in run_setup
          exec(code, locals())
        File "<string>", line 21, in <module>
      ModuleNotFoundError: No module named 'torch'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.
```
Could you guys help me to solve this problem? Thanks a lot!

**Comment by feifeibear** [comment]
It seems that you did not install torch.

**Comment by TTTanger** [comment]
> It seems that you did not install torch.
@feifeibear 

**Actually I have installed 'torch'.**

Execute **"pip list"** in the right env, the output contains torch, and the version is **2.5.1**

![1733734187568](https://github.com/user-attachments/assets/ba9c55c6-b139-4f06-a55a-6415f06eb1f0)

**Comment by feifeibear** [comment]
Could you privde your cuda driver version?

**Comment by feifeibear** [comment]
Can you use the provided docker [thufeifeibear/xdit-dev](https://hub.docker.com/r/thufeifeibear/xdit-dev)?

You can install the latest xdit in the container

**Comment by TTTanger** [comment]
@feifeibear 
I am new to Docker. Could plz guide me how to install the latest xdit by Docker? I have installed Docker and pulled your xdit docker to my server. But after I executed **"docker run --gpus all --ipc=host --ulimit memlock=-1 --ulimit stack=67108864 thufeifeibear/xdit-dev"**, I found that the image was not running.
Is my command wrong?

![1733740366661](https://github.com/user-attachments/assets/7378e053-3ad7-4089-914a-aa4b272d40a4)
